presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoPreparedStatement.java

```diff
@@ -453,7 +454,33 @@ else if (x instanceof Timestamp) {
     public void addBatch()
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "addBatch");
+        checkOpen();
+        batchValues.add(toValues(parameters));
```

Comment:
Should `execute*()` methods be disallowed once `addBatch` is called?

Reply:
@findepi Sorry, I couldn't understand well. Do you mean we should allow only `executeBatch()` and disallow other `execute*()` once `addBatch()` is used?

Reply:
i think so, but i admit i have not checked any other JDBC drivers for comparison

Reply:
@findepi I confirmed some JDBC drivers (MySQL, PostgreSQL, SQLServer & Oracle) and they don't have such limitation.

Reply:
thanks for checking!

can i
```
.setParameter(...)
.addBatch()
.setParameter(...)
.executeUpdate()
.addBatch()
.executeBatch()
```

?

in such a case, what parameters will be added in the second `addBatch()` call? (eg does `.executeUpdate()` clear current params?)

Reply:
@findepi It seems the behavior are different among Oracle and other JDBC drivers .
* Oracle throws `error occurred during batching: batch must be either executed or cleared` at executeUpdate
* MySQL, PostgreSQL & SQLServer doesn't clear at executeUpdate. 3 rows are added in this case.

Reply:
I like the "more strict" approach -- if we can opt out from supporting some niche or purely hypothetical use-case, we should.

Reply:
@findepi Added a logic to let `execute*` fail. Should we clear the status when called `clearBatch`?

Reply:
Makes sense to me. clearBatch looks like reverting the pristine state, right?

Reply:
@findepi Right. Added `isBatch = false` in clearBatch().